src/middlewares.js

```diff
@@ -424,6 +425,66 @@ export function promiseEnforceMasterKeyAccess(request) {
   return Promise.resolve();
 }
 
+export const addRateLimit = (route, id) => {
+  const config = Config.get(id);
+  const express = config.expressApp;
+  addLimitForRoute(express, route);
+};
+
+export const handleRateLimit = (api, rateLimit) => {
+  const options = Array.isArray(rateLimit) ? rateLimit : [rateLimit];
+  for (const route of options) {
+    addLimitForRoute(api, route);
+  }
+};
+
+const addLimitForRoute = (api, route) => {
+  for (const key in route) {
+    if (!RateLimitOptions[key]) {
+      throw `Invalid rate limit option "${key}"`;
+    }
+  }
+  api.use(
+    route.requestPath,
+    rateLimit({
+      windowMs: route.requestTimeWindow,
+      max: route.requestCount,
+      message: route.errorResponseMessage || RateLimitOptions.errorResponseMessage.default,
+      handler: (request, response, next, options) => {
+        response.status(429);
+        response.json({ code: Parse.Error.CONNECTION_FAILED, error: options.message });
+      },
+      skip: request => {
+        if (request.config.ip === '127.0.0.1' && !route.includeInternalRequests) {
+          return true;
```

Comment:
Like storage adapters, you need to expose a `options` section in the Parse Config options to allow a developer to customize the `rateLimit` options. For example it will allow the developer to use a Redis store (more robust than the memory store): https://www.npmjs.com/package/rate-limit-redis

2 solutions:
- add an `options` key in the `rateLimit` parse option that will be added spread here.
- add a `callback` `rateLimit` parse options like `beforeRateLimitRouteApplied(optins) {}` to allow the user to customize the rateLimit express options.

Since you have already desgined a configuration with route options, i will suggest options 1 following strategy that already exists on DatabaseAdapters like `databaseOptions`

Reply:
Thanks for the suggestion. This is not in scope for this PR. Please open a feature request after this feature has been introduced.